### PR TITLE
Implement products search

### PR DESCRIPTION
**Pull request by ChristianRimondi**

### CODE REVIEW

File: lib/solidus_graphql_api/types/ransack_json.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+          begin
+            case v
+            when String
+              value[k] = Integer(GraphQL::Schema::UniqueWithinType.decode(v).last)
+            when Array
+              v.each_with_index do |child_field, index|
+                v[index] = Integer(GraphQL::Schema::UniqueWithinType.decode(child_field.to_s).last)
+              end
+            end
+          rescue ArgumentError
```
Comment (mdesantis) [inline lib/solidus_graphql_api/types/ransack_json.rb]: do we really need to discard an exception in order to handle this flow?

File: spec/lib/solidus_graphql_api/context_spec.rb [hunk 2]
```diff
@@ -103,6 +108,21 @@
...
+  describe '#current_pricing_options' do
+    subject { schema_context.current_pricing_options }
+
+    let(:default_currency) { 'EUR' }
+    let(:country_iso) { 'IT-it' }
+    let!(:default_store) do
+      create(:store, default: true, default_currency: default_currency, cart_tax_country_iso: country_iso)
+    end
+
+    after { subject }
```
Comment (AlessioRocco) [inline spec/lib/solidus_graphql_api/context_spec.rb]: With this hook we call the subject two times in the next spec, can we avoid it by changing the last spec?

File: lib/solidus_graphql_api/context.rb [hunk 3]
```diff
@@ -37,7 +37,11 @@ def current_store
     end
 
     def current_pricing_options
-      @current_pricing_options ||= Spree::Config.pricing_options_class.from_context(self)
+      @current_pricing_options ||= if Spree.solidus_gem_version < Gem::Version.new('2.9')
```
Comment (AlessioRocco) [inline lib/solidus_graphql_api/context.rb]: What do you think to use only the "old implementation" for now, seems to me that `from_context` is a helper that doesn't do much more than calling it.

File: spec/integration/queries/products_spec.rb [hunk 4]
```diff
@@ -2,56 +2,37 @@
...
-    it { expect(option_value_nodes).to be_present }
-
-    it { expect(product_properties_nodes).to be_present }
-
-    it { expect(product_properties_property_nodes).to be_present }
-
-    it { expect(variant_nodes).to be_present }
-
-    it { expect(variant_default_price_nodes).to be_present }
+    context 'when products do not not exist' do
```
Comment (AlessioRocco) [inline spec/integration/queries/products_spec.rb]: There's a typo here.

File: spec/lib/solidus_graphql_api/queries/products_query_spec.rb [hunk 5]
```diff
@@ -3,9 +3,54 @@
...
+    described_class.new(user: user, pricing_options: pricing_options).call(query: query)
+  end
 
-  let!(:products) { create_list(:product, 2) }
+  let(:user) { create :user }
+  let(:pricing_options) { Spree::Config.pricing_options_class.new }
+  let(:query) { {} }
 
-  it { is_expected.to match_array(products) }
+  context 'when products do not exist' do
```
Comment (AlessioRocco) [inline spec/lib/solidus_graphql_api/queries/products_query_spec.rb]: We test the real behavior of this method into the integration spec, and I think we can simplify these specs by mocking the request to the Solidus search class.

File: lib/solidus_graphql_api/types/ransack_json.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+              ::Object.const_get(class_name)
+              value[k] = Integer(item_id)
+            when Array
+              v.each_with_index do |child_field, index|
+                class_name, item_id = GraphQL::Schema::UniqueWithinType.decode(child_field.to_s)
+                ::Object.const_get(class_name)
+                v[index] = Integer(item_id)
+              end
+            end
+          rescue ArgumentError, NameError
```
Comment (mdesantis) [inline lib/solidus_graphql_api/types/ransack_json.rb]: Can we move this `rescue`s around the `decode` line? Something like this:

```ruby
begin
  class_name, item_id = GraphQL::Schema::UniqueWithinType.decode(v)
rescue ArgumentError, NameError
  next
end
```

In this way we avoid rescuing exceptions that we don't want to rescue

**Pull request merged**